tools/validation-generator/src/main/java/io/spine/tools/validate/field/NestedConstraints.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.validate.field;
+
+import com.google.gson.reflect.TypeToken;
+import com.squareup.javapoet.CodeBlock;
+import io.spine.code.proto.FieldDeclaration;
+import io.spine.tools.validate.ViolationAccumulator;
+import io.spine.tools.validate.code.Expression;
+import io.spine.tools.validate.code.ViolationTemplate;
+import io.spine.validate.ConstraintViolation;
+import io.spine.validate.Validate;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A validation rule which checks that a message field does not violate its own constraints.
```

Comment:
What do you mean by "its own constraints"? 

Does this class serve the recursive definitions of messages where a message may have a field of the same type?